### ISSUE TITLE
Be able to limit append anything to elements that can be appended

### ISSUE THREAD

**Issue by philippfromme** [bpmn-io/bpmn-js-create-append-anything#30]
### Is your feature request related to a problem? Please describe.

Currently both the create and append anything popup menu list all elements that can be _created_. The append popup menu does not limit the list to elements that can be _appended_ which can will often hide the most relevant elements in the list.

Example: Event-based gateway

There are only a few elements that can be _appended_ but they're hard to find. It doesn't really make sense to offer any other options here.

![image](https://github.com/bpmn-io/bpmn-js-create-append-anything/assets/7633572/728e61cc-7bec-47a0-8552-6c1f752247c5)

### Describe the solution you'd like

Change append anything to only list elements that can be _appended_. This could also be configurable so users can decide themselves.

### Describe alternatives you've considered

Require users to add their own provider that filters the list.

### Additional context

Related to https://github.com/camunda/improved-canvas/issues/56

**Comment by nikku** [comment]
> There are only a few elements that can be appended but they're hard to find. It doesn't really make sense to offer any other options here.

@philippfromme Let's discuss this.

Append anything allows you to _add_ any element. We've received feedback / requests in the past to remove the palette entirely (which is usually far away from you). Append anything is the (only) powerful tool you usually need.

I'm all in for offering _common elements_ in a domain specific manner. You _should_ be able to append/create any relevant element (in the flow) though. 

**TLDR:** If we introduce this, then we break existing user flow, make users search for a particular element in different places, require them to understand BPMN inside out, and in the end make them not use the tool.

**Comment by philippfromme** [comment]
Based on the discussion we'll close this issue for now.

**Comment by nikku** [comment]
One thing brought up in the discussion: With the re-introduction of quick append we don't need `common` anymore.

**Comment by philippfromme** [comment]
@nikku The common group was added by `@camunda/improved-canvas` and has been removed in the meantime.

**Comment by philippfromme** [comment]
Reopening based on user feedback.

**Comment by philippfromme** [comment]
From a [user test](https://camunda.slack.com/archives/GP70M0J6M/p1714398434774069): "From an event-based gateway I should not be able to append a task", which is exactly the issue I described.
